homework10/moving circle/moving circle/ViewController.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  ViewController.swift
+//  moving circle
+//
+//  Created by Ernest Mihasenko on 2.04.22.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+    @IBOutlet weak var circleSpace: UIView!
+    @IBOutlet weak var circle: UILabel!
+    @IBOutlet weak var upButton: UIButton!
+    @IBOutlet weak var leftButton: UIButton!
+    @IBOutlet weak var downtButton: UIButton!
+    @IBOutlet weak var rightButton: UIButton!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+        circle.layer.cornerRadius = circle.bounds.size.width/2
+        circle.layer.borderWidth = 50
+    }
+
+    @IBAction func upButtonPress(_ sender: Any) {
+        if circle.layer.position.y > circleSpace.frame.minY {
```

Comment:
You need to check agaist step here: `if circle.frame.y > 5` because you subtract it later. And `circleSpace.frame` is circleSpace's parent data